### PR TITLE
Rodert fix image random cutout 2276

### PR DESCRIPTION
**Pull request by JulianRodert**
# Description

Brief Description of the PR: Bugfix for random_cutout mask which is not fully applied when randomly chosen mask location is chosen too close to the image boundary. 

Fixes # (2276)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?
Additional unit test

*  Bugfix for random_cutout mask which is not fully applied when randomly chosen mask location is chosen too close to the image boundary. Unit test applies a zero mask to 10 images of ones, and verifies the total expected sum over all images is correct. i.e. all of the zeros have been applied.

cutout_bug_fix

### CODE REVIEW

**Review by fsx950223 — APPROVED** [review]

**Review by bhack — APPROVED** [review]

**Comment by winthropharvey** [comment]
This change contradicts the cutout paper,  

```
"When cutout is applied to an im-
age, we randomly select a pixel coordinate within the image
as a center point and then place the cutout mask around that
location. This method allows for the possibility that not all
parts of the cutout mask are contained within the image. In-
terestingly, we found that allowing portions of the patches
to lay outside the borders of the image (rather than con-
straining the entire patch to be within the image) was criti-
cal to achieving good performance. Our explanation for this
phenomenon is that it is important for the model to receive
some examples where a large portion of the image is visible
during training. "
```

I'd argue that, actually, choosing the centers from only inside the image's height/length still biases the model towards increased weighting of edge pixels.  It is easy to see that central pixels, as they can be covered by patches arising from all directions, have greater odds of being covered.  The corner pixels have the least odds of being covered, as a great many coincident patches that could have arisen from outside the image are not able to be selected.  To actually have uniform odds of any given pixel being covered, the centers should be chosen from (0-half_mask_width, image_width+half_mask_width).  As this allows any possible coinciding mask on the infinite plane, even of 1 pixel coverage, to be selected, it is easy to see that all pixels have equal odds of being covered.

**Comment by bhack** [comment]
This is now in Keras-cv:

https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/random_cutout_test.py
